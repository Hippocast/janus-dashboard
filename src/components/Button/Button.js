import React from 'react'
import PropTypes from 'prop-types'

import block from '../../helpers/bem-cn'
import './Button.css'

const propTypes = {
  className: PropTypes.string,
  mod: PropTypes.oneOf([
    'primary',
    'danger',
    'default',
    'github',
    'small',
    'white'
  ]).isRequired,
  type: PropTypes.string,
  onClick: PropTypes.func,
  size: PropTypes.oneOf([
    'small'
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

const addMod = mod => (mod ? { [mod]: !!mod } : {})

const Button = ({ mod, type, onClick, children, className, size, ...props }) => {
  const b = size ? block('j-button')({ [size]: true }) : block('j-button')

  return (
    <button
      {...props}
      className={b(addMod(mod)).mix(className)()}
      type={type}
      onClick={onClick}
    >
      {children}
    </button>
  )
}

Button.propTypes = propTypes

export default Button
