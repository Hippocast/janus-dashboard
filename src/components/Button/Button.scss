@import 'index.scss';

$button-height: $hf-height-input;
$button-line-height: 1.5;
$button-font-size: $hf-font-size-base;

.j-button {
    display: flex;
    align-items: center;
    border-radius: $hf-border-radius-button;
    white-space: nowrap;
    line-height: $button-line-height;
    height: $button-height;
    font-size: $button-font-size;
    padding: $hf-padding-button;
    user-select: none;
    cursor: pointer;

    &s__wrapper {
        display: flex;
        margin-left: 10px;

        .j-button {
            margin-left: 10px;
        }

        &.tabs-nav {
            margin-left: 0;

            .j-button {
                margin-left: 0;
            }
        }
    }

    &--primary,
    &--danger,
    &--github {
        border-color: transparent;
        color: $hf-color-white;

        a {
            color: $hf-color-white;
        }
    }

    &--default,
    &--white {
        border-color: transparent;
    }

    &--default {
        color: $hf-color-black;

        a {
            color: $hf-color-black;
        }
    }

    &--white {
        color: $hf-color-button-primary;

        a {
            color: $hf-color-button-primary;
        }
    }

    &--primary {
        background-color: $hf-color-button-primary;
    }

    &--danger {
        background-color: $hf-color-button-danger;
    }

    &--github {
        background-color: $hf-color-github;
    }

    &--white {
        background-color: transparent;
    }

    &--small {
        height: ceil($button-height/2);
        line-height: ceil($button-line-height/2);
        font-size: ceil($button-font-size/2);
    }

    .j-icon {
        margin-right: 8px;
    }
}


