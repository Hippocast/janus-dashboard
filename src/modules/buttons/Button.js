import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';
import './Button.css';

const propTypes = {
    className: PropTypes.string,
    mod: PropTypes.oneOf([
        'primary',
        'danger',
        'default',
        'github',
    ]).isRequired,
    type: PropTypes.string,
    onClick: PropTypes.func,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]),
};

const b = block('j-button');

const addMod = mod => (mod ? { [mod]: !!mod } : {});

const Button = ({ mod, type, onClick, children, className }) => (
    <button
        className={b(addMod(mod)).mix(className)}
        type={type}
        onClick={onClick}
    >
        {children}
    </button>
);

Button.propTypes = propTypes;

export default Button;
