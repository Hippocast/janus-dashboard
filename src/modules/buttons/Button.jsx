import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';
import './Button.css';

const propTypes = {
  label: PropTypes.string,
  mod: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

const b = block('j-button');

const addMod = mod => mod ? { [mod]: !!mod } : {};

const Button = ({ label, mod, type, onClick, children }) => {
  const inner = children ? children : label;

  return (
    <button
      className={b(addMod(mod))}
      type={type}
      onClick={onClick}
    >
      {inner}
    </button>
  );
};

Button.propTypes = propTypes;

export default Button;
