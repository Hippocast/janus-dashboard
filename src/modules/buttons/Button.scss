@import 'index.scss';

.j-button {
    align-items: center;
    border-radius: $hf-border-radius-button;
    white-space: nowrap;
    line-height: 1.5;
    font-size: $hf-font-size-base;
    padding: $hf-padding-button;
    user-select: none;
    cursor: pointer;

    &--primary,
    &--danger {
        border-color: transparent;
        color: $hf-color-white;

        a {
            color: $hf-color-white;
        }
    }

    &--primary {
        background-color: $hf-color-button-primary;
    }

    &--danger {
        background-color: $hf-color-button-danger;
    }
}


