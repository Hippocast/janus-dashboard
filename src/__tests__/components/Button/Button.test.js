import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import Button from '../../../components/Button/Button';

describe('Button component', () => {
    const requiredProps = {
        mod: 'primary',
    };
    const renderWithRequiredProps = props => renderer.create(
        <Button
            {...requiredProps}
            {...props}
        />
    ).toJSON();

    it('renders correctly with all possible `mod`s', () => {
        const mod = [
            'primary',
            'danger',
            'default',
            'github',
            'white',
        ];

        mod.forEach(el => {
            const tree = renderer
                .create(
                    <Button
                        mod={el}
                    />
                )
                .toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    it('renders with single child', () => {
        const passedProps = {
            children: <p>mock-child</p>,
        };
        const tree = renderWithRequiredProps(passedProps);

        expect(tree).toMatchSnapshot();
    });

    it('renders with all children', () => {
        const passedProps = {
            children: [
                <p key="1">mock-child-1</p>,
                <p key="2">mock-child-2</p>,
                <p key="3">mock-child-3</p>
            ],
        };
        const tree = renderWithRequiredProps(passedProps);

        expect(tree).toMatchSnapshot();
    });

    it('renders with an additional className if passed', () => {
        const passedProps = {
            className: 'mock-class',
        };
        const tree = renderWithRequiredProps(passedProps);
        const className = tree.props.className;

        expect(className).toContain('j-button');
        expect(className).toContain(`j-button--${requiredProps.mod}`);
        expect(className).toContain('mock-class');
        expect(tree).toMatchSnapshot();
    });
});
