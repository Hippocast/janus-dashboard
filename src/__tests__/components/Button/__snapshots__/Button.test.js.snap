// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly with all possible \`mod\`s 1`] = `
<button
  className="j-button j-button--primary"
  onClick={undefined}
  type={undefined}
/>
`;

exports[`Button component renders correctly with all possible \`mod\`s 2`] = `
<button
  className="j-button j-button--danger"
  onClick={undefined}
  type={undefined}
/>
`;

exports[`Button component renders correctly with all possible \`mod\`s 3`] = `
<button
  className="j-button j-button--default"
  onClick={undefined}
  type={undefined}
/>
`;

exports[`Button component renders correctly with all possible \`mod\`s 4`] = `
<button
  className="j-button j-button--github"
  onClick={undefined}
  type={undefined}
/>
`;

exports[`Button component renders correctly with all possible \`mod\`s 5`] = `
<button
  className="j-button j-button--white"
  onClick={undefined}
  type={undefined}
/>
`;

exports[`Button component renders with all children 1`] = `
<button
  className="j-button j-button--primary"
  onClick={undefined}
  type={undefined}
>
  <p>
    mock-child-1
  </p>
  <p>
    mock-child-2
  </p>
  <p>
    mock-child-3
  </p>
</button>
`;

exports[`Button component renders with an additional className if passed 1`] = `
<button
  className="j-button j-button--primary mock-class"
  onClick={undefined}
  type={undefined}
/>
`;

exports[`Button component renders with single child 1`] = `
<button
  className="j-button j-button--primary"
  onClick={undefined}
  type={undefined}
>
  <p>
    mock-child
  </p>
</button>
`;
